pkg/proxy/ipvs/proxier.go

```diff
@@ -892,10 +897,15 @@ func (proxier *Proxier) OnServiceSynced() {
 	proxier.mu.Lock()
 	proxier.servicesSynced = true
 	proxier.setInitialized(proxier.endpointSlicesSynced)
+	proxier.updateWeights = true
 	proxier.mu.Unlock()
 
 	// Sync unconditionally - this is called once per lifetime.
 	proxier.syncProxyRules()
+
```

Comment:
Remove the `proxier.updateWeights = false` from here and instead do it where it's checked (see below)